gammapy/data/pointing.py

```diff
@@ -67,45 +70,194 @@ class FixedPointingInfo:
+
+            self._fixed_altaz = fixed_altaz
+            self._fixed_icrs = None
+        else:
+            raise ValueError(f"Unsupported pointing mode for FixedPointingInfo: {mode}")
+
+    @classmethod
+    def from_fits_header(cls, header):
+        obs_mode = header.get("OBS_MODE", "POINTING")
+        mode = PointingMode.from_gadf_string(obs_mode)
+        try:
+            location = earth_location_from_dict(header)
+        except KeyError:
+            location = None
+
+        # we allow missing RA_PNT / DEC_PNT in POINTING for some reason...
+        # FIXME: actually enforce this to be present instead of using nan
+        ra = u.Quantity(header.get("RA_PNT", np.nan), u.deg)
+        dec = u.Quantity(header.get("DEC_PNT", np.nan), u.deg)
+        alt = header.get("ALT_PNT")
+        az = header.get("AZ_PNT")
+        legacy_altaz = None
+
+        fixed_icrs = None
+        pointing_altaz = None
+
+        # we can be more strict with DRIFT, as support was only added recently
+        if mode is PointingMode.DRIFT:
+            if alt is None or az is None:
+                raise IOError(
+                    "Keywords ALT_PNT and AZ_PNT are required for OBSMODE=DRIFT"
+                )
+            pointing_altaz = AltAz(alt=alt * u.deg, az=az * u.deg)
+        else:
+            fixed_icrs = SkyCoord(ra, dec)
+            # store given altaz also for POINTING for backwards compatibility,
+            # FIXME: remove in 2.0
+            if alt is not None and az is not None:
+                legacy_altaz = AltAz(alt=alt * u.deg, az=az * u.deg)
+
+        time_start = header.get("TSTART")
+        time_stop = header.get("TSTOP")
+        time_ref = None
+
+        if time_start is not None or time_stop is not None:
+            time_ref = time_ref_from_dict(header)
+            time_unit = u.Unit(header.get("TIMEUNIT", "s"), format="fits")
+
+            if time_start is not None:
+                time_start = time_ref + u.Quantity(time_start, time_unit)
+
+            if time_stop is not None:
+                time_stop = time_ref + u.Quantity(time_stop, time_unit)
+
+        return cls(
+            mode=mode,
+            location=location,
+            fixed_icrs=fixed_icrs,
+            fixed_altaz=pointing_altaz,
+            time_start=time_start,
+            time_stop=time_stop,
+            time_ref=time_ref,
+            legacy_altaz=legacy_altaz,
+        )
+
+    def to_fits_header(self, format="gadf", version="0.3", time_ref=None):
+        if format != "gadf":
+            raise ValueError(f'Only the "gadf" format supported, got {format}')
+
+        if version not in {"0.2", "0.3"}:
+            raise ValueError(f"Unsupported version {version} for format {format}")
+
+        if self.mode == PointingMode.DRIFT and version == "0.2":
+            raise ValueError("mode=DRIFT is only supported by GADF 0.3")
+
+        header = fits.Header()
+        if self.mode == PointingMode.POINTING:
```

Comment:
Could move this code to the `PointingMode` class directly...would lead to a shorter implementation here...

Reply:
What could be moved? I don't see what part of the code is dealing with the `PonitingMode` itself, the `PointingMode` just decides which information stored in `PointingInfo` is put into the header.

Reply:
Oh yes, you are right. As we already have `PointingMode.from_gadf_string()` I thought it might make sense to have `PointingMode.to_fits_header()` as well, but it would contain effectively only two lines of code and the returned header would be barely useful. So fine as is...